pandas/core/generic.py

```diff
@@ -3353,10 +3353,15 @@ def _maybe_update_cacher(self, clear=False, verify_is_copy=True):
             if ref is None:
                 del self._cacher
             else:
+                # Note: we need to call ref._maybe_cache_changed even in the
+                #  case where it will raise.  (Uh, not clear why)
                 try:
                     ref._maybe_cache_changed(cacher[0], self)
-                except Exception:
-                    pass
+                    assert ref.ndim == self.ndim, (ref.ndim, self.ndim)
+                except AssertionError:
```

Comment:
does this raise in the test suite?

Reply:
moved the first assertion to the `else` block to clarify that it is not the assertion that is raising.

There _are_ places in the test suite where the `ref._maybe_cache_changed` call raises an `AssertionError` because of a shape mismatch, but AFAICT checking for that situation and only calling it in non-raising cases breaks several tests